[ML] AIOps: Tweak log rate changes in log rate analysis results table.

walterra opened the pull request:
## Summary

Part of #187684.

This moves functions related to log rate changes to the `@kbn/aiops_log_rate_analysis` package.

- `getLogRateAnalysisType` was renamed to `getLogRateAnalysisTypeForHistogram` to indicate its use with histogram data.
- `getLogRateAnalysisTypeForCounts` was added for cases where we don't have the histogram data available but just the doc counts for baseline an deviation time ranges. This isn't used yet as of this PR but will be in a follow up in combination with the o11y AI assistant.
- `getSwappedWindowParameters` is a helper to consolidate inline code that's used to swap baseline and deviation when we detected a dip in log rate.
- Rounding for the log rate change messages was tweaked. Changes below `10x` will now be rounded to one digit to avoid messages like `1x increase`.
- Tweaked/Shortened the message for 0 in baseline or deviation to just `45 up from 0 in baseline` / `down to 0 from 45 in baseline`.

### Checklist

- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [x] This was checked for breaking API changes and was [labeled appropriately](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

elasticmachine commented:
Pinging @elastic/ml-ui (:ml)

peteharverson commented on x-pack/packages/ml/aiops_log_rate_analysis/get_log_rate_change.ts:
```diff
@@ -0,0 +1,82 @@
+        message: i18n.translate(
+          'xpack.aiops.logRateAnalysis.resultsTableGroups.logRateDocIncreaseLabel',
+          {
+            defaultMessage: '{deviationBucketRate} up from 0 in baseline',
```
Or `up to {deviationBucketRate} from 0 in baseline` for consistency with the dip version?

peteharverson reviewed (approved):
Other than a suggestion for a slight edit to the text, tested and LGTM

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/222604)
* Commit: 1ff0b8c451afc2b5450e46be018f62ac6613bb16

### Failed CI Steps
* [Quick Checks](https://buildkite.com/elastic/kibana-pull-request/builds/222604#0190cb1c-ad33-4f69-a7e6-20c1cc0b587e)

cc @walterra

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/222604","number":222604,"commit":"1ff0b8c451afc2b5450e46be018f62ac6613bb16"}],"number":222604}
buildkite-pr-comment-->

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/222629)
* Commit: 62585b47f66293bd8fec708ae83ba416abfa34d4

### Failed CI Steps
* [Jest Tests #11](https://buildkite.com/elastic/kibana-pull-request/builds/222629#0190cb70-a4e1-49fb-9c44-1443f752fc78)
* [Jest Tests #11](https://buildkite.com/elastic/kibana-pull-request/builds/222629#0190cb8d-6667-4faa-8539-ace664e5c67d)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/222629#0190cb70-a4e1-49fb-9c44-1443f752fc78) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/222629/jobs/0190cb70-a4e1-49fb-9c44-1443f752fc78/artifacts/0190cb8d-4b47-4024-a1d8-a377c14727b8) Jest Tests #<span></span>11 / getLogRateChange handles a baseline rate of 0 without throwing an error
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/222629#0190cb8d-6667-4faa-8539-ace664e5c67d) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/222629/jobs/0190cb8d-6667-4faa-8539-ace664e5c67d/artifacts/0190cb92-b10e-48b2-be22-22b63d090aa5) Jest Tests #<span></span>11 / getLogRateChange handles a baseline rate of 0 without throwing an error

### History
* :broken_heart: [Build #222604](https://buildkite.com/elastic/kibana-pull-request/builds/222604) failed 1ff0b8c451afc2b5450e46be018f62ac6613bb16

cc @walterra

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/222629","number":222629,"commit":"62585b47f66293bd8fec708ae83ba416abfa34d4"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/222604","number":222604,"commit":"1ff0b8c451afc2b5450e46be018f62ac6613bb16"}],"number":222629}
buildkite-pr-comment-->

walterra commented:
I noticed a bug how the log rate change text was calculated and updated. Because we always calculated it based on the window parameters selected in the chart, if you moved the brushes after the analysis was done, it would also update the text in the log rate change columns. I fixed it by moving `currentAnalysisWindowParameters` to the redux state to make it available for the columns so the calculation is always done on the values corresponding on what has been analysed. For clarity, I renamed the other `windowParameters` to `chartWindowParameters` to make the difference more obvious.

Before:

![aiops-log-rate-analysis-column-update-0001](https://github.com/user-attachments/assets/440509b7-50a2-496e-89b0-c7d6fcf846af)

After:

![aiops-log-rate-analysis-column-update-0002](https://github.com/user-attachments/assets/1b8ab36a-2a10-4379-b612-8788c8bdb5c6)

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/222665)
* Commit: 244c96baf57c382678648f97829fca17cba8e9f0

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/222665#0190cba1-a8f4-42fc-b375-1af6512c74af)

### History
* :broken_heart: [Build #222629](https://buildkite.com/elastic/kibana-pull-request/builds/222629) failed 62585b47f66293bd8fec708ae83ba416abfa34d4
* :broken_heart: [Build #222604](https://buildkite.com/elastic/kibana-pull-request/builds/222604) failed 1ff0b8c451afc2b5450e46be018f62ac6613bb16

cc @walterra

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/222665","number":222665,"commit":"244c96baf57c382678648f97829fca17cba8e9f0"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/222629","number":222629,"commit":"62585b47f66293bd8fec708ae83ba416abfa34d4"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/222604","number":222604,"commit":"1ff0b8c451afc2b5450e46be018f62ac6613bb16"}],"number":222665}
buildkite-pr-comment-->

alvarezmelissa87 commented on x-pack/packages/ml/aiops_log_rate_analysis/api/stream_reducer.ts:
```diff
@@ -80,7 +85,12 @@ export const logRateAnalysisResultsSlice = createSlice({
       state.significantItemsGroups = [];
     },
-    resetAll: () => getDefaultState(),
+    // reset the results but keeps
```
Nit: unfinished comment explanation?

walterra replied:
Thanks for spotting this! Fixed in [6941aa7](https://github.com/elastic/kibana/pull/188648/commits/6941aa72590acfd86cb2d8b1e6b5a1189a17abd5).

alvarezmelissa87 reviewed (approved):
LGTM ⚡
